build_tools/cmake/iree_check_test.cmake

```diff
@@ -241,13 +245,15 @@ function(iree_check_single_backend_test_suite)
 
   # Omit tests for which the specified driver or target backend is not enabled.
   # This overlaps with directory exclusions and other filtering mechanisms.
-  string(TOUPPER ${_RULE_DRIVER} _UPPERCASE_DRIVER)
-  string(REPLACE "-" "_" _NORMALIZED_DRIVER ${_UPPERCASE_DRIVER})
-  if(NOT DEFINED IREE_HAL_DRIVER_${_NORMALIZED_DRIVER})
-    message(SEND_ERROR "Unknown driver '${_RULE_DRIVER}'. Check IREE_HAL_DRIVER_* options.")
-  endif()
-  if(NOT IREE_HAL_DRIVER_${_NORMALIZED_DRIVER})
-    return()
+  if(DEFINED _RULE_DRIVER)
```

Comment:
Seems redundant? You check if it's defined above

Reply:
I'm using `iree_check_single_backend_test_suite`, which uses `iree_check_test` under the hood.

`iree_check_single_backend_test_suite` here tests if `IREE_HAL_DRIVER_{_RULE_DRIVER}` exists and is enabled. Since `_RULE_DRIVER` is not defined, I want to skip over that test.

The test above in the file is in `iree_check_test`, right before defining the `iree_native_test` that uses `iree-check-module`.